blockchain/vm/instructions.go

```diff
@@ -921,6 +906,21 @@ func opPush1(pc *uint64, evm *EVM, scope *ScopeContext) ([]byte, error) {
 	return nil, nil
 }
 
+func opSelfdestruct(pc *uint64, evm *EVM, scope *ScopeContext) ([]byte, error) {
+	if evm.interpreter.readOnly {
+		return nil, ErrWriteProtection
+	}
```

Comment:
Maybe this conditional exists in `interpreter.go:264` `if in.readOnly { if operation.writes.. }`?
Setting `JumpTable[SELFDESTRUCT].writes = true` might be sufficient.

Reply:
Oh, nevermind. geth moved away from centralized `operation.writes` check to per-instruction check.